libraries/AP_BLHeli/AP_BLHeli.cpp

```diff
@@ -1361,9 +1361,16 @@ void AP_BLHeli::init(void)
      */
 #if APM_BUILD_TYPE(APM_BUILD_Rover)
     AP_MotorsUGV *motors = AP::motors_ugv();
+    if (motors) {
+        uint16_t motormask = motors->get_motor_mask();
```

Comment:
This doesn't look really right to me - the otype logic above is wrong for rover still and the masks get fiddled with below again, which also seems wrong. I'm prepared to believe it kind of works but the logic is really hard to follow.

I think you need to add a static function to both AP_Motors and AP_MotorsUGV to do this check and that at least makes this code more logical. We probably also need an AP_Motors::to_rcout_pwm_type() or something to replace the otype logic above.